Sources/TripKitUI/helper/RxTripKit/TKServer+Rx.swift

```diff
@@ -116,20 +116,27 @@ extension Reactive where Base: TKServer {
           }
           
           let hitAgain: TKServer.RepeatHandler?
-          let model = try? result.get()
```

Comment:
@nighthawk Like an earlier case, there might be a reason why you swallowed the throw here, but it's creating some problems. 

Our back end has put in place a restriction on when a new bundle can be purchased, and testing it on the app when the error from the throw is swallowed, this is what we get,

![Simulator Screen Shot - iPhone 11 - 2022-01-21 at 22 57 33](https://user-images.githubusercontent.com/2722800/150523957-afc35e2e-fbb2-4ace-b6c0-9beb1bf89c95.png)

Because the error is swallowed, the `model` is nil, and that when propagating upwards via the `onNext` leads to ServerError error 0, which means no data. I can see why this is right, because it is in fact, no data.

But if we don't swallow the error and having the subscriber propagates the error upwards, we will get

![Simulator Screen Shot - iPhone 11 - 2022-01-21 at 22 58 33](https://user-images.githubusercontent.com/2722800/150524157-bc2e7757-e6f5-4fa7-960b-dc1d80b466c8.png)

thanks for us converting the below response

```
{
    "usererror": true,
    "errorCode": 482,
    "title": "maas.bundle.purchase.error",
    "error": "Purchasing another pass more than 7 days before current pass expires is not allowed"
}
```

to an error via `TKServer.error...`, the error is more easily understandable

Of course, I can also deal with this by having the end caller in `BundleDetailsViewModel` pick up the 482 status code our back end returns, but that means app must be updated for any future cases. If we were to pass the error upwards, at least the user can still get some understandable error messages, even w/o updating the app (sure, the app still needs to be updated once to have this change, but not the subsequent ones).

Anyways, putting this here for you to take a look as I'm not confident that this won't break anything, considering you had a reason previously to use `try?` rather than `try`

Reply:
Yeah, same reason here as in #140. There might not be any data returned by the backend, just a status code. Re-throwing the "no data" error would then break that. The new `/agenda` endpoints rely on this, for example. So this code should similarly check what kind of error is returned and not pass on a `noData` error.

Reply:
Hmm, another issue is that this is the `stream` method. That isn't meant to fail at all by itself. Instead the caller should use the `repeatHandler` to stop hitting. Say, you're fetching real-time updates every 5 seconds and in the middle, you have a network drop out, you just want to ignore that one that failed, rather than stopping the sequence when it hits one error.